### PR TITLE
Implement team domain workflows

### LINKED ISSUE #102: Implement team domain workflows

### LINKED ISSUE #107: Terraform update for deploy

### PR DESCRIPTION
**Pull request by paulobressan**
Close #102 
Close #107

### CODE REVIEW

File: examples/config/rpc.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ db_path="dev.db"
 crds_path="examples/crds"
 secret="fabric@txpipe"
 topic="events"
+invite_ttl_min="15"
```
Comment (gonzalezzfelipe) [inline examples/config/rpc.toml:6]: Is this supposed to be here?

File: src/domain/event/mod.rs [hunk 2]
```diff
@@ -61,6 +61,28 @@ pub struct ProjectSecretCreated {
...
 into_event!(ProjectSecretCreated);
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProjectUserInviteCreated {
+    pub id: String,
+    pub project_id: String,
+    pub email: String,
+    pub role: String,
+    pub code: String,
+    pub expire_in: DateTime<Utc>,
```
Comment (gonzalezzfelipe) [inline src/domain/event/mod.rs:71]:
```suggestion
    pub expires_in: DateTime<Utc>,
```

File: src/domain/event/mod.rs [hunk 3]
```diff
@@ -202,6 +235,30 @@ mod tests {
...
     }
+    impl Default for ProjectUserInviteCreated {
+        fn default() -> Self {
+            Self {
+                id: Uuid::new_v4().to_string(),
+                project_id: Uuid::new_v4().to_string(),
+                email: "p@txpipe.io".into(),
+                code: "123".into(),
+                role: ProjectUserRole::Owner.to_string(),
+                expire_in: Utc::now() + Duration::from_secs(15 * 60),
```
Comment (gonzalezzfelipe) [inline src/domain/event/mod.rs:246]:
```suggestion
                expires_in: Utc::now() + Duration::from_secs(15 * 60),
```

File: src/domain/project/mod.rs [hunk 4]
```diff
@@ -126,15 +140,115 @@ impl From<ProjectSecretCreated> for ProjectSecret {
...
-#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct ProjectUserInvite {
+    pub id: String,
+    pub project_id: String,
+    pub email: String,
+    pub code: String,
+    pub role: ProjectUserRole,
+    pub status: ProjectUserInviteStatus,
+    pub expire_in: DateTime<Utc>,
```
Comment (gonzalezzfelipe) [inline src/domain/project/mod.rs:151]:
```suggestion
    pub expires_in: DateTime<Utc>,
```

File: src/domain/project/mod.rs [hunk 5]
```diff
@@ -126,15 +140,115 @@ impl From<ProjectSecretCreated> for ProjectSecret {
...
+
+    fn try_from(value: ProjectUserInviteCreated) -> std::result::Result<Self, Self::Error> {
+        Ok(Self {
+            id: value.id,
+            project_id: value.project_id,
+            email: value.email,
+            role: value.role.parse()?,
+            code: value.code,
+            status: ProjectUserInviteStatus::Sent,
+            expire_in: value.expire_in,
```
Comment (gonzalezzfelipe) [inline src/domain/project/mod.rs:166]:
```suggestion
            expires_in: value.expires_in,
```

File: src/domain/project/mod.rs [hunk 6]
```diff
@@ -174,8 +288,24 @@ mod tests {
...
+    impl Default for ProjectUserInvite {
+        fn default() -> Self {
+            Self {
+                id: Uuid::new_v4().to_string(),
+                project_id: Uuid::new_v4().to_string(),
+                email: "p@txpipe.io".into(),
+                role: ProjectUserRole::Owner,
+                code: "123".into(),
+                status: ProjectUserInviteStatus::Sent,
+                expire_in: Utc::now() + Duration::from_secs(15 * 60),
```
Comment (gonzalezzfelipe) [inline src/domain/project/mod.rs:305]:
```suggestion
                expires_in: Utc::now() + Duration::from_secs(15 * 60),
```

File: src/driven/cache/migrations/20240606_tables.sql [hunk 7]
```diff
@@ -25,11 +25,25 @@ CREATE TABLE IF NOT EXISTS resource (
...
 );
 
+CREATE TABLE IF NOT EXISTS project_user_invite (
+  id TEXT PRIMARY KEY NOT NULL,
+  project_id TEXT NOT NULL,
+  email TEXT NOT NULL,
+  role TEXT NOT NULL,
+  code TEXT NOT NULL,
+  status TEXT NOT NULL,
+  expire_in DATETIME NOT NULL,
```
Comment (gonzalezzfelipe) [inline src/driven/cache/migrations/20240606_tables.sql:41]:
```suggestion
  expires_in DATETIME NOT NULL,
```

File: src/driven/cache/project.rs [hunk 8]
```diff
@@ -323,6 +327,106 @@ impl ProjectDrivenCache for SqliteProjectDrivenCache {
...
+        let invite = sqlx::query_as::<_, ProjectUserInvite>(
+            r#"
+                SELECT 
+                    pui.id,
+                    pui.project_id,
+                    pui.email,
+                    pui."role",
+                    pui.code,
+                    pui.status,
+                    pui.expire_in,
```
Comment (gonzalezzfelipe) [inline src/driven/cache/project.rs:341]:
```suggestion
                    pui.expires_in,
```

File: src/driven/cache/project.rs [hunk 9]
```diff
@@ -323,6 +327,106 @@ impl ProjectDrivenCache for SqliteProjectDrivenCache {
...
+        sqlx::query!(
+            r#"
+                INSERT INTO project_user_invite(
+                    id, 
+                    project_id,
+                    email,
+                    "role",
+                    code,
+                    status,
+                    expire_in,
```
Comment (gonzalezzfelipe) [inline src/driven/cache/project.rs:368]:
```suggestion
                    expires_in, 
```

File: src/driven/cache/project.rs [hunk 10]
```diff
@@ -323,6 +327,106 @@ impl ProjectDrivenCache for SqliteProjectDrivenCache {
...
+                )
+                VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9);
+            "#,
+            invite.id,
+            invite.project_id,
+            invite.email,
+            role,
+            invite.code,
+            status,
+            invite.expire_in,
```
Comment (gonzalezzfelipe) [inline src/driven/cache/project.rs:380]:
```suggestion
            invite.expires_in,
```

File: src/driven/cache/project.rs [hunk 11]
```diff
@@ -361,10 +465,38 @@ impl FromRow<'_, SqliteRow> for ProjectSecret {
...
+            project_id: row.try_get("project_id")?,
+            email: row.try_get("email")?,
+            code: row.try_get("code")?,
+            role: role
+                .parse()
+                .map_err(|err: Error| sqlx::Error::Decode(err.into()))?,
+            status: status
+                .parse()
+                .map_err(|err: Error| sqlx::Error::Decode(err.into()))?,
+            expire_in: row.try_get("expire_in")?,
```
Comment (gonzalezzfelipe) [inline src/driven/cache/project.rs:497]:
```suggestion
            expires_in: row.try_get("expires_in")?,
```

**Review by gonzalezzfelipe — APPROVED** [review]

File: examples/config/rpc.toml (hunk 1 above)
Comment (paulobressan) [reply examples/config/rpc.toml:6]: No, I forgot it there. I will remove

File: bootstrap/rpc/config.tf [hunk 12]
```diff
@@ -21,6 +21,8 @@ resource "kubernetes_config_map_v1" "fabric_rpc_config" {
         auth0_client_secret = var.auth0_client_secret
         auth0_audience      = var.auth0_audience
         stripe_api_key      = var.stripe_api_key
+        email_ses_access_key_id = var.email_ses_access_key_id
+        email_ses_secret_access_key = var.email_ses_secret_access_key
```
Comment (gonzalezzfelipe) [inline bootstrap/rpc/config.tf:25]: Inside this `templatefile` function, you have to define the parameters needed to template the file. In this case, you are missing some:
```suggestion
        email_ses_access_key_id = var.email_ses_access_key_id
        email_ses_secret_access_key = var.email_ses_secret_access_key
        email_invite_ttl_min = var.email_invite_ttl_min
        email_ses_region = var.email_ses_region
        email_ses_verified_email = var.email_ses_verified_email
```

**Review by gonzalezzfelipe — APPROVED** [review]
The terraform is missing formatting, you can format the files by doing `terraform fmt path_to_file`

File: bootstrap/rpc/config.tf (hunk 12 above)
Comment (paulobressan) [reply bootstrap/rpc/config.tf]: Do they need to be there even using default values in the main file?

Comment (paulobressan) [reply bootstrap/rpc/config.tf]: ignore it, you are right!

**Review by gonzalezzfelipe — APPROVED** [review]

**Pull request merged**